accounts-db/src/accounts_db.rs

```diff
@@ -12561,33 +12561,40 @@ pub mod tests {
 
     #[test]
     fn test_hash_stored_account() {
-        // This test uses some UNSAFE tricks to detect most of account's field
-        // addition and deletion without changing the hash code
+        // This test uses some UNSAFE tricks to detect most of hashing code changes, resulting from
+        // account's field additions and deletions of StoredAccountMeta and AccountSharedData and
+        // hashing-order changes.
 
         const ACCOUNT_DATA_LEN: usize = 3;
-        // the type of InputTuple elements must not contain references;
+        // the type of InputFields elements must not contain references;
         // they should be simple scalars or data blobs
-        type InputTuple = (
+        // repr(C) is needed for abi-stability in the dirtiest variant of std::mem::transmute().
+        #[repr(C)]
+        struct InputFields(
```

Comment:
I'm a bit surprised that plain tuple and tuple struct's abi differ; but i guess this is a life. ;)

Reply:
In case you were curious about the the layout change for this specific tuple, like I was last night, I logged the first byte of each tuple field in 1.69.0 and 1.71.0+:

```
slot_int, meta_int, account_meta_int, data_int, offset_int, hash_int
1.69.0: 112 160 96 40 104 37
1.71.0: 104 96 160 40 48 37
```
Note the underlying data array bytes are in reverse order: [160..1].

So in memory the layout order is:
```
1.69.0: [meta, slot, offset, account_meta, data, hash]
1.71.0: [account_meta, slot, meta, offset, data, hash] 
```
both still take 160 bytes, but just have a different order 🤷‍♂️